doc/tutorial/web/rest/asab.webservice.md

```diff
@@ -0,0 +1,271 @@
+    
+    if __name__ == __main__
+        app = CRUDApp()   
+        app.run()
+        
+#### ./myrestapi/app.py
+
+Here, define our CRUDApp object, we will first need some imports:   
+ 
+    import asab
+    import asab.web
+    import asab.web.rest
+  
+Now we need to add some default configs:    
+ 
+    asab.Config.add_defaults(
+	{
+		'asab:storage': {
+			'type': 'mongodb',
+			'mongodb_uri': 'mongodb://mongouser:mongopassword@mongoipaddress:27017',
+			'mongodb_database': 'mongodatabase'
+		},
+	})
+    
+Next, we describe the class, it inherits from the basic ASAB Application class, but we need to expand 
+it a little (see comments in the code for more details):
+    
+    
+    class CRUDApp(asab.Application):
+
+	    def __init__(self):
+		    super().__init__()
+		    # Circumvent the need for -w flag when starting the app
+		    import asab.storage
+		    self.add_module(asab.web.Module)
+		    self.add_module(asab.storage.Module)
+
+		    # Locate the web service
+		    self.WebService = self.get_service("asab.WebService")
+		    self.WebContainer = asab.web.WebContainer(self.WebService, "web")
+		    self.WebContainer.WebApp.middlewares.append(asab.web.rest.JsonExceptionMiddleware)
+
+		    # Initialize services, we can initialize one, or several API handlers/services here
+		    from .crud.handler import CRUDWebHandler
+		    from .crud.service import CRUDService
+		    self.CRUDService = CRUDService(self)
+		    # We need to pass the CRUDService as an argument, when instantiating the class
+		    self.CRUDWebHandler = CRUDWebHandler(self, self.CRUDService)
+
+This could also be a BSPUMP.Application, enabling us to include a bspump pipeline in our microservice. 
+To do this, inherit instead from bspump.BSPumpApplication and add service, connections and pipelines as 
+usual.
+   
+#### ./myrestapi/crud/handler.py
+
+The handler is where we define the operations our microservice will be able to perform. As the 
+name suggests, here we handle incoming requests and return the results back to the caller. It is strongly suggested, 
+that we do these CRUD methods one by one and test them straight away (the way we do this, we create a Handler method for 
+creating first, then we create its Service method right after, and finally we test using Postman), if you did not set 
+up your database test instance yet, now will be time to do it. If you struggle, check section below, where we use docker 
+to setup a simple MongoDB containered instance. Now without further ado, lets jump into creating the handler.  
+As usual, we start by importing modules:   
+
+    import asab
+    import asab.web.rest
+    
+    
+Next is the CRUDWebhandler class:
+
+    class CRUDWebHandler(object):
+	def __init__(self, app, mongo_svc):
+		self.CRUDService = mongo_svc
+
+		web_app = app.WebContainer.WebApp
+		
+		web_app.router.add_put('/crud-api/{collection}', self.create) # Create endpoint url   
```

Comment:
`web_app.router.add_get('/crud-api/{collection}', self.read)` = Get many 

I also suggest to organize this to separate Read and Write methods.

Reply:
"This" meaning the endpoints?